src/peripheral/mod.rs

```diff
@@ -302,22 +302,44 @@ pub struct MPU {
 #[repr(C)]
 pub struct NVIC {
     /// Interrupt Set-Enable
-    pub iser: [RW<u32>; 8],
-    reserved0: [u32; 24],
+    pub iser: [RW<u32>; 16],
+    reserved0: [u32; 16],
     /// Interrupt Clear-Enable
-    pub icer: [RW<u32>; 8],
-    reserved1: [u32; 24],
+    pub icer: [RW<u32>; 16],
+    reserved1: [u32; 16],
     /// Interrupt Set-Pending
-    pub ispr: [RW<u32>; 8],
-    reserved2: [u32; 24],
+    pub ispr: [RW<u32>; 16],
+    reserved2: [u32; 16],
     /// Interrupt Clear-Pending
-    pub icpr: [RW<u32>; 8],
-    reserved3: [u32; 24],
+    pub icpr: [RW<u32>; 16],
+    reserved3: [u32; 16],
     /// Interrupt Active Bit
-    pub iabr: [RO<u32>; 8],
-    reserved4: [u32; 56],
+    pub iabr: [RO<u32>; 16],
+    reserved4: [u32; 48],
+
     /// Interrupt Priority
-    pub ipr: [RW<u8>; 240],
+    ///
+    /// On ARMv7-M, 124 word-sized registers are available. Each of those
+    /// contains of 4 interrupt priorities of 8 byte each.The architecture
+    /// specifically allows accessing those along byte boundaries, so they are
+    /// represented as 496 byte-sized registers, for convenience, and to allow
+    /// atomic priority updates.
+    ///
+    /// On ARMv6-M, the registers must only be accessed along word boundaries,
+    /// so convenient byte-sized representation wouldn't work on that
+    /// architecture.
+    #[cfg(not(armv6m))]
+    pub ipr: [RW<u8>; 496],
+    #[cfg(armv6m)]
+    // The compiler lets itself be confused by all the `#[cfg(...)]` around
+    // here, and complains about missing documentation for `ipr` when compiling
+    // for ARMv6-M. The following directive works around that.
+    // I hope that rustdoc correctly picks up on the documentation, but wasn't
+    // able to verify that, due to a Cargo bug[1]. As it stands, the
+    // documentation always shows the ARMv7-M variant of the field.
+    // [1] https://github.com/rust-lang/cargo/issues/2539
+    #[allow(missing_docs)]
```

Comment:
Does anyone know if there's a rust issue for the problem I've described here? I couldn't find one.